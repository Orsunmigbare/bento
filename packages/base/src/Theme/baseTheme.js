import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
