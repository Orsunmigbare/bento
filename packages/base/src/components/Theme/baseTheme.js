export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
