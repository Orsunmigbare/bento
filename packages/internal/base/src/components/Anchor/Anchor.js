import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { commonPropTypes, themePropTypes } from "../../constants/themeContants"
import { applyVariants } from "../../theming"

const Anchor = ({ children, className, ...restProps }) => {
  return (
    <AComponent className={className} {...restProps}>
      {children}
    </AComponent>
  )
}

Anchor.themeProps = {
  a: {
    name: "A",
    description: "Link",
    type: themePropTypes.style
  }
}

export const AComponent = styled.a`
  text-decoration: none;
  color: currentColor;

  &:hover,
  &:active,
  &:focus {
    text-decoration: none;
    color: currentColor;
  }

  ${applyVariants(Anchor.themeProps.a)};
`

Anchor.defaultProps = {
  className: ""
}

Anchor.propTypes = {
  children: commonPropTypes.children.isRequired,
  className: PropTypes.string
}

export default Anchor
