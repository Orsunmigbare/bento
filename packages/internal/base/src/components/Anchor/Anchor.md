<div>
  <A href="https://nrk.no/">Tester</A>
</div>
