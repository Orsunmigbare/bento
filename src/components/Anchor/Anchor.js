import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { commonPropTypes, themePropTypes } from "../../constants/themeContants"
import { applyVariants } from "../../theming"

const Anchor = ({ children, className, ...restProps }) => (
  <AComponent className={className} {...restProps}>
    {children}
  </AComponent>
)

Anchor.themeProps = {
  anchor: {
    name: "Anchor",
    description: "HTML anchor link",
    type: themePropTypes.style
  }
}

export const AComponent = styled.a`
  text-decoration: none;
  color: currentColor;

  &:hover,
  &:active,
  &:focus {
    text-decoration: none;
    color: currentColor;
    outline: none;
  }

  ${applyVariants(Anchor.themeProps.anchor)};
`

Anchor.defaultProps = {
  className: "",
  href: ""
}

Anchor.propTypes = {
  children: commonPropTypes.children.isRequired,
  /**
   * URL prop
   */
  href: PropTypes.string,
  className: PropTypes.string
}

export default Anchor
