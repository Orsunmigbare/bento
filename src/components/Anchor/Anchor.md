```jsx harmony
A

<div>
  <A href="https://nrk.no/">Tester</A>
</div>
```
