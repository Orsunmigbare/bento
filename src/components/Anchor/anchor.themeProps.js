import { themePropTypes } from "../../theming"

const themeProps = {
  anchor: {
    name: "Anchor",
    description: "HTML anchor link",
    type: themePropTypes.style
  }
}
export default themeProps
