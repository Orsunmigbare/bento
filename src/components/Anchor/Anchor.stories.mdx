import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Anchor from "./Anchor"

<Meta title="Components|Anchor" component={Anchor}/>


# Anchor
Anchor is an HTML tag, that is used to define the beginning and end of a hypertext link.
Anchor helps search engines to determine the subject matter of the destination URL.
Users can click on the tag to react the link target, like with a normal `<a>` -tag 


# import

```javascript
import {Anchor} from "@staccx/bento"
```

# Types of Anchors


## Standard Anchor
This is the standard Anchor

<Preview>
  <Story name="Standard">
   <Anchor href="https://www.stacc.com">Check out our page!</Anchor>
  </Story>
</Preview>


## Inline anchor
An achor tag will place itself inline with other text making it possible to create links
inside text

<Preview>
  <Story name="Inline">
  <p>
    this text contains an Anchor tag, that you can click to get more 
    information about the subject, 
   <Anchor href="https://www.stacc.com"> check out our page!</Anchor>
  </p>
  </Story>
</Preview>


## Multiple anchors
Multiple anchor tags will also place themselves inline with the other tags.

<Preview>
  <Story name="Multiple">
    <>
    <Anchor href="https://www.stacc.com">This is a link</Anchor>
    <Anchor href="https://www.stacc.com">This is another</Anchor>
    <Anchor href="https://www.stacc.com">this is a third </Anchor>
  </>
  </Story>
</Preview>


# Props for Anchor
<Props of={Anchor} />
