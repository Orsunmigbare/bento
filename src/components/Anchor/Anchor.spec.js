import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../theming/themes/baseTheme"
import Anchor from "./Anchor"

const Child = () => <p>Body</p>

describe("Anchor", () => {
  describe("Snapshots", () => {
    it("no props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Anchor>
              <Child />
            </Anchor>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("href", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Anchor href="https://stacc.com">
              <Child />
            </Anchor>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("no children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Anchor href="https://stacc.com" />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
