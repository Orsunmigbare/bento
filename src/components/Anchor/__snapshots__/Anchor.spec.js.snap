// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Snapshots href 1`] = `
<a
  className="sc-AykKC jICarN"
  href="https://stacc.com"
>
  <p>
    Body
  </p>
</a>
`;

exports[`Anchor Snapshots no children 1`] = `
<a
  className="sc-AykKC jICarN"
  href="https://stacc.com"
/>
`;

exports[`Anchor Snapshots no props 1`] = `
<a
  className="sc-AykKC jICarN"
  href=""
>
  <p>
    Body
  </p>
</a>
`;
